python/tank/authentication/user.py

```diff
@@ -138,4 +251,10 @@ def deserialize_user(payload):
 
     :returns: A ShotgunUser derived instance.
     """
-    return ShotgunUser(user_impl.deserialize_user(payload))
+    impl = user_impl.deserialize_user(payload)
+
+    # We use the presence of cookies as an indicator that we are using SSO.
+    if isinstance(impl, user_impl.SessionUser) and impl.get_cookies() is not None:
+        return ShotgunSamlUser(impl)
```

Comment:
@jfboismenu are we ready on the desktop end to handle this correctly? Last time you identified some issues...